tests/api_fastapi/core_api/routes/public/test_task_instances.py

```diff
@@ -1127,3 +1127,283 @@ def test_should_respond_dependencies_mapped(self, test_client, session):
             "print_the_context/0/dependencies",
         )
         assert response.status_code == 200, response.text
+
+
+class TestGetTaskInstancesBatch(TestTaskInstanceEndpoint):
+    @pytest.mark.parametrize(
+        "task_instances, update_extras, payload, expected_ti_count",
+        [
+            pytest.param(
+                [
+                    {"queue": "test_queue_1"},
+                    {"queue": "test_queue_2"},
+                    {"queue": "test_queue_3"},
+                ],
+                True,
+                {"queue": ["test_queue_1", "test_queue_2"]},
+                2,
+                id="test queue filter",
+            ),
+            pytest.param(
+                [
+                    {"executor": "test_exec_1"},
+                    {"executor": "test_exec_2"},
+                    {"executor": "test_exec_3"},
+                ],
+                True,
+                {"executor": ["test_exec_1", "test_exec_2"]},
+                2,
+                id="test executor filter",
+            ),
+            pytest.param(
+                [
+                    {"duration": 100},
+                    {"duration": 150},
+                    {"duration": 200},
+                ],
+                True,
+                {"duration_gte": 100, "duration_lte": 200},
+                3,
+                id="test duration filter",
+            ),
+            pytest.param(
+                [
+                    {"execution_date": DEFAULT_DATETIME_1},
```

Comment:
Why do we have mix of execution_date and logic_date?

Reply:
Yes this is not great. The reason is because this dictionary is used to instantiate a `TI` and create it in the database. (With `setattr(attr, value)` construct). `logical_date` do not exist on the TI ORM model.

This is coming from the legacy code, maybe there is a better way now to set the `logical_date` on a task instance without using the association proxy `execution_date` ?

But then the next dictionary is actually the request payload, send by the client, which exposes a `logical_date` and not the deprecated `execution_date`.

Reply:
Indeed I was behind, https://github.com/apache/airflow/pull/43902 has been merged and `execution_date` is 💀